lib/logger/lib/logger.ex

```diff
@@ -573,7 +573,7 @@ defmodule Logger do
     if compare_levels(level, min_level) != :lt do
       macro_log(level, data, metadata, caller)
     else
-      :ok
+      handle_unused_variable_warnings(data)
```

Comment:
> Wouldn't that partially defeat the purpose of compile time purging?

I think the main purpose of purging is to have very detailed log in development but much less noise in production, and also remove needless pressure from the `Logger`.
I assume values for logging are already computed before the `Logger` call or needs to be called like in example above.
If we want to purge right hand side completely, maybe we can provide configuration option for this, or do this for `(fn -> "Done: #{cleanup()}" end)` calls. :bowtie: